test_infra/src/lib.rs

```diff
@@ -722,17 +670,17 @@ impl Guest {
         Guest {
             tmp_dir,
             disk_config,
-            fw_path: Self::fw_path(FwType::RustHypervisorFirmware),
+            fw_path: fw_path.unwrap_or_default(),
```

Comment:
Ok by looking at the usage of `fw_path` from the `Guest` structure, I think we should remove it from the structure. The firmware can be directly created from `tests/integration.rs`, no need to be passed to the `Guest` structure, since 99% of the time, the value of `fw_path` is needed from within `tests/integration.rs`.
I think there's only one case that needs some special handling: `api_create_body()`. Simply add a new `fw_path` parameter to it.

Reply:
Good advise. Will update the PR accordingly.